at_client/src/main/java/org/atsign/client/cli/Register.java

```diff
@@ -13,90 +14,138 @@
         }
-        String registrarUrl = configReader.getProperty("registrar", "url");
+
+        registrarUrl = isRegistrarV3 ? configReader.getProperty("registrarV3", "url")
+                : configReader.getProperty("registrar", "url");
         if (registrarUrl == null) {
-            //reading config from older configuration syntax for backwards compatability
+            // reading config from older configuration syntax for backwards compatability
             registrarUrl = configReader.getProperty("REGISTRAR_URL");
         }
-        String apiKey = configReader.getProperty("registrar", "apiKey");
+
         if (apiKey == null) {
-            //reading config from older configuration syntax for backwards compatability
-            apiKey = configReader.getProperty("API_KEY");
+            try {
+                apiKey = configReader.getProperty("registrar", "apiKey");
+            } catch (Exception e) {
+                // reading config from older configuration syntax for backwards compatability
+                apiKey = configReader.getProperty("API_KEY");
+            }
         }
 
         if (rootDomain == null || rootPort == null || registrarUrl == null || apiKey == null) {
-            System.err.println("Please make sure to set all relevant configuration in src/main/resources/config.yaml");
+            System.err.println(
+                    "Please make sure to set all relevant configuration in src/main/resources/config.yaml");
             System.exit(1);
         }
 
-        Scanner scanner = new Scanner(System.in);
-        RegisterUtil registerUtil = new RegisterUtil();
+        if (!isRegistrarV3) {
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("Getting free atsign");
+            atsign = new AtSign(registerUtil.getFreeAtsign(registrarUrl, apiKey));
+            System.out.println("Got atsign: " + atsign);
 
-        System.out.println("Getting free atsign");
-        AtSign atsign = new AtSign(registerUtil.getFreeAtsign(registrarUrl, apiKey));
-        System.out.println("Got atsign: " + atsign);
+            System.out.println("Sending one-time-password to :" + email);
+            if (registerUtil.registerAtsign(email, atsign, registrarUrl, apiKey)) {
+                System.out.println("Enter OTP received on: " + email);
 
-        System.out.println("Sending one-time-password to :" + email);
-        if (registerUtil.registerAtsign(email, atsign, registrarUrl, apiKey)) {
-            System.out.println("Enter OTP received on: " + email);
+                otp = scanner.nextLine();
+                System.out.println("Validating one-time-password");
+                validationResponse = registerUtil.validateOtp(email, atsign, otp, registrarUrl, apiKey, false);
+                // if validationResponse is retry, the OTP entered is incorrect. Ask user to
+                // re-enter correct OTP
+                if ("retry".equals(validationResponse)) {
```

Comment:
if we have finite list of validation response, have those in an enum e.g retry, follow-up